addon/session-stores/cookie.js

```diff
@@ -198,5 +257,14 @@ export default BaseStore.extend({
     } else {
       return RSVP.resolve();
     }
+  },
+
+  // FIXME: maybe be causing getter bugs with rewriteCookie called before
+  // setter can return value
```

Comment:
According to the guides, the setter should return the value you want to assign
```
fullName: Ember.computed('firstName', 'lastName', {
    set(key, value) {
      var [firstName, lastName] = value.split(/\s+/);
      this.set('firstName', firstName);
      this.set('lastName',  lastName);
      this.rewriteCookie();
      return value;
    }
  })
```
An example buggy situation is where there's `this.get('fullName')` inside `rewriteCookie()`. It's a non issue if this code is wrong.